src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -195,6 +197,60 @@ public TutorialClass findTutorialClassFromList(TutorialClass tutorialClass, Modu
         return tutorialClassInList;
     }
 
+    /**
+     * Randomly allocates the students in {@code tutorial class} into {@code numOfTeams} of different teams.
+     *
+     * @param moduleCode of the tutorial class.
+     * @param tutorialClass to allocate the different students into the teams to.
+     * @param numOfTeams of teams to split into.
+     */
+    public void randomTeamAllocation(ModuleCode moduleCode, TutorialClass tutorialClass, int numOfTeams) {
+        requireAllNonNull(moduleCode, tutorialClass, numOfTeams);
+        ArrayList<TutorialTeam> teams = tutorialClass.getTeams();
+        ArrayList<Person> classList = tutorialClass.getStudents();
+        int classSize = tutorialClass.getStudents().size();
+        int teamSize = (int) Math.ceil((double) classSize / numOfTeams);
+        teams.clear();
+
+        // creating the teams to add into
+        for (int i = 1; i <= numOfTeams; i++) {
+            String teamName = "Team" + i;
+            TutorialTeam team = new TutorialTeam(teamName, teamSize);
+            addTeam(tutorialClass, team);
+        }
+
+        Random random = new Random();
+        int k = 0;
+        while (k < classList.size()) {
+            int randInt = random.nextInt(numOfTeams);
+            TutorialTeam currTeam = tutorialClass.getTeams().get(randInt);
+            if (hasTeamSizeExceeded(currTeam)) {
+                continue;
+            }
+
+            Person student = classList.get(k);
+            if (!isSamePersonInTeam(student, currTeam)) {
+                currTeam.addStudent(student);
+                k++;
+            }
+        }
+    }
+
+    /**
+     * Checks if the same person is in the team already.
+     * @param student to check if exist in team.
+     * @param tutorialTeam to check.
+     * @return a boolean true if the student exists in the team, else false.
+     */
+    public boolean isSamePersonInTeam(Person student, TutorialTeam tutorialTeam) {
```

Comment:
IMO, any functions that are related to the team (in this case both checking for isSamePersonInTeam and hasTeamExceededSize) should be both placed in the TutorialTeam class instead since it accesses the variables in the team class. We should maintain the abstraction to the class that it only affects. AddressBook should only contains methods that needs interaction between various components